torch_xla/csrc/tensor.cpp

```diff
@@ -136,35 +136,59 @@ std::vector<xla::util::Cleanup> LockDevices(const std::set<Device>& devices) {
   return unlocker;
 }
 
-struct TensorHasher {
```

Comment:
What are the changes to this file for? They don't look related to the bool tensor.

Reply:
They started as bool tensors, and my "Let me try that on TPU".
Then hell broke lose and I accumulated them here.

Reply:
I'd rather split into two commits if it's not too hard.

Reply:
Please ...

Reply:
I don't think this is an unreasonable ask, but ok.